### PR TITLE
Made the port optional

### LINKED ISSUE #238: C#: Add port autodetection and make `port` parameter optional
### Feature Description
We should have feature parity between different MineStat editions. The Python (and other) editions made the `port` parameter optional, if it can be automatically detected, e.g. through SRV records or default ports.

- [ ] Add port autodetection for default ports
- [ ] Make `port` parameter optional
- [ ] Document new usage patterns in README and docstring

> Originally proposed by @nimeng1299 in issue #237

### PR DESCRIPTION
**Pull request by smellilac**
Fixes #238

### CODE REVIEW

File: CSharp/MineStat/MineStat.cs [hunk 1]
```diff
@@ -30,6 +30,15 @@
 using System.Xml;
 using System.Xml.Linq;
 using System.Xml.XPath;
+using System.Threading.Tasks;
+using System.Net;
+
+#if NETSTANDARD2_0
```
Comment (ldilley) [inline CSharp/MineStat/MineStat.cs:36]: Would we want to specify `NETSTANDARD2_0_OR_GREATER` here?

File: CSharp/MineStat/MineStat.cs [hunk 2]
```diff
@@ -30,6 +30,15 @@
...
+using DnsClient;
+#elif NET46
```
Comment (ldilley) [inline CSharp/MineStat/MineStat.cs:38]: Similarly, would we want to use `NET46_OR_GREATER` here?

File: CSharp/MineStat/MineStat.cs [hunk 3]
```diff
@@ -30,6 +30,15 @@
...
+using ARSoft.Tools.Net;
+using ARSoft.Tools.Net.Dns;
+#endif
```
Comment (ldilley) [inline CSharp/MineStat/MineStat.cs:41]: Would we want to handle all other cases with an `#else` here?

File: CSharp/MineStat/MineStat.cs [hunk 4]
```diff
@@ -231,11 +240,55 @@ public MineStat(string address, ushort port, int timeout = DefaultTimeout, SlpPr
...
+                null);
+
+            var srvRecords = dnsMessage.AnswerRecords.OfType<SrvRecord>();
+
+            if (srvRecords.Any())
+            {
+                var srvRecord = srvRecords.First();
+                return (ushort)srvRecord.Port;
+            }
+            #endif
```
Comment (ldilley) [inline CSharp/MineStat/MineStat.cs:277]: Are there any conditions that should be covered under an `#else`?

**Review by ldilley — CHANGES_REQUESTED** [review]
Thank you for the contribution, @smellilac. Aside from my review questions, the only other issue I see is with indentation. Our code style uses 2 spaces.

I am also including @Ajoro to see if he has any additional suggestions.

File: CSharp/MineStat/MineStat.cs (hunk 3 above)
Comment (smellilac) [reply CSharp/MineStat/MineStat.cs:41]: I don't see how `#else` can be used logically here. The only thing that comes to mind is a construction like this:

```
#else
#error This target framework is not supported by this code. Please ensure the target is .NET Standard 2.0 or greater, or .NET Framework 4.6 or greater.
```

**Comment by ldilley** [comment]
Closing due to lack of response from author.

**Pull request closed without merging**